### ISSUE TITLE
Conservative whitespace collapse misses some spaces

### ISSUE THREAD

**Issue by stevenvachon** [posthtml/htmlnano#34]
Before:
```html
<head>
<style>body{color:red}</style>
<style></style>
</head>
<body>
```

After:
```html
<head> <style>body{color:red}</style>  </head> <body>
```

Notice the two spaces after `</style>`.

Whitespace collapse should take place after element removals, if it isn't already.

**Comment by stevenvachon** [comment]
Another issue directly related to this one is trailing new lines get converted to spaces when they should probably be stripped completely.

Before:
```html
<html>
</html>

```
After:
```html
<html> </html> 
```
(you'll need to select the text to notice the trailing space)

**Comment by maltsev** [comment]
Fixed in #24.
